org.sugarlabs.MusicBlocks.json

```diff
@@ -0,0 +1,125 @@
+    "finish-args": [
+        "--share=network",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--device=dri",
+        "--filesystem=home",
+        "--env=MUSICBLOCKS_PATH=/app/share/musicblocks"
+    ],
+    "build-options" : {
+        "env": {
+            "NPM_CONFIG_LOGLEVEL": "info"
+        }
+    },
+    "modules": [
+        {
+            "name": "nodejs",
+            "cleanup": [
+                "/include",
+                "/share",
+                "/app/lib/node_modules/npm/changelogs",
+                "/app/lib/node_modules/npm/doc",
+                "/app/lib/node_modules/npm/html",
+                "/app/lib/node_modules/npm/man",
+                "/app/lib/node_modules/npm/scripts"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://nodejs.org/dist/v14.7.0/node-v14.7.0.tar.xz",
+                    "sha256": "ca2f1c63f3f2bf22247d7386bfc31e0295caa953f39f7079210170a886288e6f"
+                }
+            ]
+        },
+        {
+            "name": "mb-launcher",
+            "buildsystem": "simple",
+            "build-commands": [
+                "python3 setup.py install --prefix=${FLATPAK_DEST}"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/sugarlabs/musicblocks-launcher",
+                    "tag": "v1.1.0",
+                    "commit": "1808e64149691690f8db5f16d70eebfdde5bc340"
+                }
+            ]
+        },
+        {
+            "name": "musicblocks",
+            "buildsystem": "simple",
+            "build-commands": [
+                "rm -rf musicblocks/po",
+                "rm -rf musicblocks/usermanual",
+                "rm -rf musicblocks/documentation*",
+                "rm -rf musicblocks/guide*",
+                "rm -rf musicblocks/charts",
+                "rm -rf musicblocks/mouse-art",
+                "rm -rf musicblocks/screenshots",
+                "rm musicblocks/package*",
+                "mkdir /app/share/musicblocks",
+                "cp -ra musicblocks/* /app/share/musicblocks"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/sugarlabs/musicblocks",
+                    "tag": "v3.2",
+                    "commit": "010d7f90cd68ac25c3c07886f1f5955ef2f196dc",
+                    "dest": "musicblocks"
+                },
+                {
+                    "type": "patch",
+                    "path": "save-interface.patch"
+                }
+            ]
+        },
+        {
+            "name": "electron-app",
+            "build-options": {
+                "env": {
+                    "electron_config_cache": "/run/build/electron-app/flatpak-node/npm-cache"
+                },
+                "build-args": [ "--share=network" ]
```

Comment:
That's not going to work, Flathub builds are offline. Look at https://github.com/flatpak/flatpak-builder-tools/blob/master/node/README.md to work around it.

If it's about just some component not being stored in generated-sources.json, @refi64 might know what's going on.

Reply:
@barthalion My main issue is that it seems to be trying to redownload the electron binary during postinstall. The binary is specified in `generated-sources.json`, and the path is stored in `electron_config_cache`, but it still tries to download it again anyway (resulting in `ELIFECYCLE` and `EAI_AGAIN` errors if it's not connected to the network). Is there something I can do to force it to use the cached binary?